File: 3447-clear-digits/3447-clear-digits.java
public class Solution {

    public String clearDigits(String s) {
        int answerLength = 0;
        char[] charArray = s.toCharArray();

        for (int charIndex = 0; charIndex < s.length(); charIndex++) {
            if (Character.isDigit(s.charAt(charIndex))) {
                answerLength = Math.max(answerLength - 1, 0);
            } else {
                charArray[answerLength++] = s.charAt(charIndex);
            }
        }
        s = new String(charArray, 0, answerLength);

        return s;
    }
}

File: 3447-clear-digits/3447-clear-digits.cpp
class Solution {
public:
    string clearDigits(string s) {
        
        int answerLength = 0;

        for (int charIndex = 0; charIndex < s.size(); charIndex++) {
            if (isdigit(s[charIndex])) {
            
                answerLength = max(answerLength - 1, 0);
            } else {
                s[answerLength++] = s[charIndex];
            }
        }

        s.resize(answerLength);

        return s;
    }
};